packages/@aws-cdk/aws-ec2/lib/bastion-host.ts

```diff
@@ -0,0 +1,73 @@
+import { PolicyStatement } from "@aws-cdk/aws-iam";
+import { CfnOutput, Construct } from "@aws-cdk/core";
+import { AmazonLinuxGeneration, AmazonLinuxImage, InstanceClass, InstanceSize, InstanceType } from ".";
+import { Instance } from "./instance";
+import { ISecurityGroup } from "./security-group";
+import { IVpc, SubnetType } from "./vpc";
+
+/**
+ * Properties of the bastion host
+ */
+export interface BastionHostLinuxProps {
+
+  /**
+   * In which AZ to place the instance within the VPC
+   *
+   * @default - Random zone.
+   */
+  readonly availabilityZone?: string;
+
+  /**
+   * VPC to launch the instance in.
+   */
+  readonly vpc: IVpc;
+
+  /**
+   * The name of the instance
+   *
+   * @default - CDK generated name
+   */
+  readonly instanceName?: string;
+
+  /**
+   * Use public subnet instead of private one
+   *
+   * @default - false
+   */
+  readonly publicSubnets?: boolean;
```

Comment:
Shouldn't this be a `subnetSelection`?

And shouldn't public subnets be the default? What use is a Bastion host in a private subnet?

EDIT: Oh I see, it probably has to do with whether you  intend to access the bastion directly using SSH or using Session Manager. Could you at least make that clear in the documentation text? Or better yet, maybe it should be a property to select between the two?

Reply:
I think there are three scenarios:

1. Public access using SSH -> public subnet
2. Access via SSM -> private subnet
3. Access using SSH from on-prem -> private subnet via VPN/DC

So public subnets should not be used imho...

Reply:
Re use cases, we use all of these and also have use case where we have had to have a baston host, (jump box) in all 3 zones.

When conducting penetration tests we will place a (bastion) host in each zone (public, private, ..), sometime the only way to test a zone get to jump through bastions in each zone  using proxy chains.
So in my imho this construct should be able to live in any zone, just default to public.

Reply:
Also you should be ideally able to place this in an explicit subnet.

Reply:
Yes, so I still feel we should expose the `subnetSelection` (and have it default to something sane, and I think `PRIVATE` is the best sane default we can have). It would immediately benefit from future extensions we would add to the `subnetSelection` mechanism (for example, selecting individual subnets such as @slipdexic requested).  And it should have explicit documentation on there saying you need to select public subnets if you want to access it from the internet.

Would it make sense to have a top-level helper method for a common use case, `bastion.allowSshAccessFrom(peer, peer, ...)` or something? The only thing it would do is encapsulate port 22, but it might be just a little nicer.

Also, @hoegertn, could you add a section to the README about the 2 Bastion Host use cases (SSM access and SSH access)?

Reply:
Sounds good. I will change it accordingly.